Warning in weakptr.c in master branch (clang, hpcgap)

Another warning is here:
```
libtool: compile:  gcc -MQ obj/weakptr.lo -MMD -MP -MF obj/.deps/weakptr.d -m64 -Qunused-arguments -pthread -Werror -I./hpcgap -I. -DHAVE_CONFIG_H -I/Volumes/hudson-fs/workspace/GAP-newbuild-compilers/GAPCOPTS/64build/GAPREADLINE/noreadline/HPCMODE/hpcgap/label/clang-xcode/GAP-master-snapshot/extern/install/gmp/include -I/Volumes/hudson-fs/workspace/GAP-newbuild-compilers/GAPCOPTS/64build/GAPREADLINE/noreadline/HPCMODE/hpcgap/label/clang-xcode/GAP-master-snapshot/extern/install/gc/include -I/Volumes/hudson-fs/workspace/GAP-newbuild-compilers/GAPCOPTS/64build/GAPREADLINE/noreadline/HPCMODE/hpcgap/label/clang-xcode/GAP-master-snapshot/extern/install/libatomic_ops/include -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -c gen/weakptr.c  -fno-common -DPIC -o obj/.libs/weakptr.o
hpcgap/src/weakptr.c:178:16: error: unused variable 'list2' [-Werror,-Wunused-variable]
  volatile Obj list2 = list;
               ^
hpcgap/src/weakptr.c:276:16: error: unused variable 'val2' [-Werror,-Wunused-variable]
  volatile Obj val2 = val;
               ^
2 errors generated.
make: *** [obj/weakptr.lo] Error 1
```
Happens when GAP is compiled with clang on macOS with hpcgap enabled.

--- a/src/weakptr.c
+++ b/src/weakptr.c
@@ -183,7 +183,7 @@ Obj FuncWeakPointerObj( Obj self, Obj list ) {
   for (i = 1; i <= len ; i++) 
     { 
 #ifdef BOEHM_GC
-      Obj tmp = ELM0_LIST(list, i);
+      Obj tmp = ELM0_LIST(list2, i);
       ELM_WPOBJ(wp,i) = tmp;
       if (IS_BAG_REF(tmp))
         REGISTER_WP(&ELM_WPOBJ(wp, i), tmp);
@@ -309,11 +309,11 @@ Obj FuncSetElmWPObj(Obj self, Obj wp, Obj pos, Obj val)
   MEMBAR_READ();
   if (IS_BAG_REF(tmp) && ELM_WPOBJ(wp, ipos))
     FORGET_WP(&ELM_WPOBJ(wp, ipos));
-#endif
+  ELM_WPOBJ(wp,ipos) = val2;
+  if (IS_BAG_REF(val2))
+    REGISTER_WP(&ELM_WPOBJ(wp, ipos), val2);
+#else
   ELM_WPOBJ(wp,ipos) = val;
-#ifdef BOEHM_GC
-  if (IS_BAG_REF(val))
-    REGISTER_WP(&ELM_WPOBJ(wp, ipos), val);
 #endif
   CHANGED_BAG(wp);
   return 0;

